libusb 1.0 conversion fix and n950 identification support

sre opened the pull request:
Hi Pali,

Here are two small patches. One adds support for "0xFFFF -I" together with N950. Without this patch one could already flash the kernel via USB (I did not try anything else) but one could not identify the device using "-I". The other fixes a copy/paste mistake in my libusb1.0 patch, which has been found by Aaro.

-- Sebastian

pali commented:
USB endpoint fix is in master (by different commit).

For adding support for new device you need to modify also usb-device.c file. And ideally also local.c (but this is not required for usb support). Run git grep DEVICE_RX_51 and you will see...

sre commented:
Actually the additional information is only used to generate the device string for the USB IDs. I added the N950 device string to all relevant USB devices, that I could see with nokia's flashing tool and added minimal support for MKII rev. 108 used by my n950.

pali commented:
I finished usb device detection, so you will need to add similar checks for N950. Currently there are merge conflicts.

And for mkii rather set dev->data |= ( 1 << 30 ) when sw_release command is supported (instead addding new field into usb device structure).

pali commented:
Hi Sebastian, are you going to update/fix this pull request?

pali commented:
I pushed commits for N950 support. Check if it is working...
